lib/library.rb

```diff
@@ -0,0 +1,53 @@
+require 'date'
+require 'pry'
+
+class Library
+  attr_accessor :books #TODO duedate person status
+
+  def initialize(options={})
+    @books = options[:books] || []
+  end
+
+  def check_in(book)
+    add = {title: book[:title],
+           author: book[:author],
+           status: :okay,
+           person: :non,
+           duedate: " "}
+    @books.push add
+  end
+
+  def lend(user, args={})
+    book = @books.detect { |obj| obj[args.keys.first] == args[args.keys.first] }
+    case
+      when book[:status] == :lended then
+        'Sorry, book is already with another reader'
+      when user.portfolio.select { |a| a[:duedate] > Date.today.strftime('%F') } != [] then
```

Comment:
Metrics/LineLength: Line is too long. [91/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/CaseIndentation: Indent `when` as deep as `case`. (https://github.com/bbatsov/ruby-style-guide#indent-when-to-case)